Source/Windows/Common/VolatileMetadata.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "Common/VolatileMetadata.h"
+
+#include <FEXCore/fextl/unordered_map.h>
+#include <FEXCore/fextl/string.h>
+#include <FEXCore/fextl/set.h>
+#include <FEXCore/Utils/IntervalList.h>
+
+namespace FEX::Windows {
+inline void ApplyFEXExtendedVolatileMetadata(FEX::VolatileMetadata::ExtendedVolatileMetadata& ExtendedMetaData, fextl::set<uint64_t>& VolatileInstructions, FEXCore::IntervalList<uint64_t>& VolatileValidRanges, uint64_t Address, uint64_t EndAddress) {
+  // Load FEX extended volatile metadata.
+  // Walk the volatile instructions first if they exist.
+  for (const auto it_inst : ExtendedMetaData.VolatileInstructions) {
+    if ((it_inst + Address) < EndAddress) {
```

Comment:
Sorry if I missed the explanation, but is this check needed for anything?

Reply:
> partial mappings, bad user input, application updated

Seems like disabling all extended metadata for this module and printing a warn log is a better approach here. Particularly if the application was updated, we'll effectively mark a random set of instructions otherwise. I could envision similarly dangerous effects for partial mappings, and "bad user input" seems like something to warn about either way if we're checking for it already.

Reply:
Added a log and early exited, but removed what we already added to the set is hard/adds overhead. So let it be on the developer at that point to fix their worldview of memory coherency.